code/game/objects/structures/depot.dm

```diff
@@ -113,6 +113,8 @@
 		depotarea.updateicon()
 
 	for(var/obj/structure/closet/L in range(30, T))
+		for(var/obj/O in L)
+			qdel(O)
 		L.open()
```

Comment:
I suspect the 4 lines above may not be necessary; I suspect that what's happening is that the closet opening, before the explosion, together with how explosion process, it leaving "gaps". 

If this is all gone, when explosions happen, it *should* result in all the contents getting `ex_act`'d. The fact the lockers don't open should prevent the atmos from pushing them around and causing weird issues.

Reply:
Having tested this by spawning lockers, I found that leaving out this code caused some locker items to survive, e.g. grenades that armed, etc. So it does seem necessary.